source/adapters/level_zero/context.hpp

```diff
@@ -141,9 +141,9 @@ struct ur_context_handle_t_ : _ur_object {
   // head.
   //
   // Cache of event pools to which host-visible events are added to.
-  std::vector<std::list<ze_event_pool_handle_t>> ZeEventPoolCache{4};
+  std::vector<std::list<ze_event_pool_handle_t>> ZeEventPoolCache{8};
   std::vector<std::unordered_map<ze_device_handle_t, size_t>>
-      ZeEventPoolCacheDeviceMap{4};
+      ZeEventPoolCacheDeviceMap{8};
```

Comment:
2 EventPoolCacheDeviceMap reserved for each: hostVisible, not hostvisible, device and hostvisible and device and not hostvisible. Since we can't query the ext flags of eventpooldesc from eventpoolhandle, we have to reserve the space in case of each of these situations.

Reply:
What if you need profiling on top of that?
Which pool would you choose?

If you want to permute all combination this would quickly be out of control.
- host visible
- counter based for immediate
- counter based for non immediate
- profiling

we are already at 16